transport/src/main/java/io/netty/channel/DefaultChannelConfig.java

```diff
@@ -334,18 +334,8 @@ private boolean isAutoFlush() {
      * @param autoFlush Auto flush value to set for the option.
      *
      * @return {@code this}
-     *
-     * @throws UnsupportedOperationException If {@link EventLoop} for the channel is not an instance of
-     * {@link SingleThreadEventLoop}
      */
     private ChannelConfig setAutoFlush(boolean autoFlush) {
-        // Auto-flush requires a change in API of EventLoop which can not be done in 4.1, so the SingleThreadEventLoop
-        // is the only implementation that supports it.
-        if (autoFlush && channel.isRegistered() && !(channel.eventLoop() instanceof SingleThreadEventLoop)) {
```

Comment:
I feel we should fail-fast instead of silently converting writes to `writeAndFlush()`. This was just protecting from a change post-channel registration though.